util/types/time.go

```diff
@@ -1103,38 +1074,88 @@ func ParseDateFromNum(num int64) (Time, error) {
 	return ParseTimeFromNum(num, mysql.TypeDate, MinFsp)
 }
 
-func checkDatetime(t Time) bool {
-	if t.IsZero() {
-		return true
+func checkDateType(t TimeInternal) error {
+	year, month, day := t.Year(), t.Month(), t.Day()
+	if year == 0 && month == 0 && day == 0 {
```

Comment:
they are different: checkDateRange check date in range 1000-01-01 to 9999-12-31
while here check whether the date is ZeroDate.